java/src/main/native/ai_onnxruntime_OrtSession_SessionOptions.c

```diff
@@ -615,3 +616,50 @@ JNIEXPORT void JNICALL Java_ai_onnxruntime_OrtSession_00024SessionOptions_addROC
     throwOrtException(jniEnv,convertErrorCode(ORT_INVALID_ARGUMENT),"This binary was not compiled with ROCM support.");
   #endif
 }
+
+/*
+ * Class::    ai_onnxruntime_OrtSession_SessionOptions
+ * Method:    appendExecutionProvider
+ * Signature: (JILjava/lang/String)V
+ */
+JNIEXPORT void JNICALL Java_ai_onnxruntime_OrtSession_00024SessionOptions_sessionOptionsAppendExecutionProvider(
+    JNIEnv* jniEnv, jobject jobj, jlong apiHandle, jlong optionsHandle,
+    jstring jepName, jobjectArray configKeyArr, jobjectArray configValueArr) {
+  (void)jobj;
+  char errMsgBuf[512];
+  const char* epName = (*jniEnv)->GetStringUTFChars(jniEnv, jepName, NULL);
+  if (strcmp(epName, "XNNPACK") == 0) {
```

Comment:
perhaps we don't need to check the EP name here, it's also done in the C API function.
https://github.com/microsoft/onnxruntime/blob/0c6037b5abe571fc43a55ef7a9d2f846820fbe5d/onnxruntime/core/session/provider_registration.cc#L76-L80

Reply:
If this is a generic system and ORT will throw a sensible exception if you pass in an unexpected or not compiled in EP then the strcmp & the ifndef can go away.

Reply:
Thanks for reminding. It sounds goods